hathor/p2p/peer.py

```diff
@@ -50,6 +56,38 @@ class PeerFlags(str, Enum):
     RETRIES_EXCEEDED = 'retries_exceeded'
 
 
+def _parse_entrypoint(entrypoint_string: str) -> Entrypoint:
+    entrypoint = Entrypoint.parse(entrypoint_string)
+    if entrypoint.peer_id is not None:
+        raise ValueError('do not add id= to peer.json entrypoints')
+    return entrypoint
+
+
+def _parse_pubkey(pubkey_string: str) -> rsa.RSAPublicKey:
+    public_key_der = base64.b64decode(pubkey_string)
+    public_key = serialization.load_der_public_key(data=public_key_der, backend=default_backend())
+    assert public_key is not None
+    return public_key
+
+
+def _parse_privkey(privkey_string: str) -> rsa.RSAPrivateKeyWithSerialization:
+    private_key_der = base64.b64decode(privkey_string)
+    private_key = serialization.load_der_private_key(data=private_key_der, password=None, backend=default_backend())
+    assert private_key is not None
+    return private_key
+
+
+def _calculate_peer_id(public_key: rsa.RSAPublicKey) -> PeerId:
+    public_der = public_key.public_bytes(
+        encoding=serialization.Encoding.DER,
+        format=serialization.PublicFormat.SubjectPublicKeyInfo,
+    )
+    h1 = hashlib.sha256(public_der)
+    h2 = hashlib.sha256(h1.digest())
+    return PeerId(h2.digest())
+
+
+@dataclass
 class Peer:
```

Comment:
Should we rename it to `BasePeer` and create an `UnverifiedPeer`?

Reply:
A `BasePeer` is not needed anymore. I split off some info into a `PeerInfo` class. But I haven't renamed `Peer` to `UnverifiedPeer`. What about `ReceivedPeer`? Or we can rename both the peer and the storage to `UnverifiedPeer` and `UnverifiedPeerStorage` (Instead of `ReceivedPeerStorage`).

Reply:
Done.

Reply:
Thanks!